File: app/src/main/java/com/robotnec/budget/core/persistence/dao/BaseDao.java
package com.robotnec.budget.core.persistence.dao;

import java.util.List;

/**
 * @author zak <zak@swingpulse.com>
 */

public interface BaseDao<T> {

    T findById(long id);

    List<T> getAll();

    boolean remove(long id);

    boolean createOrUpdate(T entity);
}

File: app/src/main/java/com/robotnec/budget/core/persistence/dao/BaseDao.kt
package com.robotnec.budget.core.persistence.dao

/**
 * @author zak zak@swingpulse.com>
 */
interface BaseDao<T> {

    val all: List<T>

    fun findById(id: Long): T

    fun remove(id: Long): Boolean

    fun createOrUpdate(entity: T): Boolean
}
